src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -36,10 +37,38 @@ public class Shredder {
 
   private final DocumentLimitsConfig documentLimits;
 
+  private final JsonShreddingMetricsReporter jsonShreddingMetricsReporter;
```

Comment:
No, cannot be assigned like this -- it is stateful object, we need new instance for every call.